include/rmm/device_buffer.hpp

```diff
@@ -383,6 +383,22 @@ class device_buffer {
    *-------------------------------------------------------------------------**/
   mr::device_memory_resource* memory_resource() const noexcept { return _mr; }
 
+  /**--------------------------------------------------------------------------*
+   * @brief Copies rmm::device_buffer to a preallocated host buffer.
+   *-------------------------------------------------------------------------**/
+  void copy_to_host(void* host_buffer) const {
```

Comment:
@jrhemstad wants this to be a free function, not a device_buffer member function.

Reply:
Also should accept a stream parameter (default to 0) and use cudaMemcpyAsync().

Reply:
> ...wants this to be a free function, not a device_buffer member function.

Is there a reason to prefer a function over a method? Happy to go either way. Just curious is all.

> Also should accept a stream parameter (default to 0) and use cudaMemcpyAsync().

Ok have tried to handle this. Please take a look.

Reply:
Defer to @jrhemstad on the method vs. function question.

Reply:
In general, unless something needs access to private internal state of an object, it's better to use a free function. The number of member functions should be kept to a minimal set that provide only the functionality necessary to use the object.

Reply:
I also suspect this function will go away if/when we add a `host_buffer`. So better to keep it independent of the `device_buffer` class definition.

Reply:
Yep, that makes sense. Tried to resolve this in the latest commit. Please take a look 🙂